### DISCUSSION TITLE
ch6-cloud/ch6-05-load-balance

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#602] [General]
# ch6-cloud/ch6-05-load-balance

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch6-cloud/ch6-05-load-balance.html

**Comment by ZBIGBEAR** [comment]
((6/7)*(6/7))^7≈0.11554334736，也就是选择第一个节点的概率是11.55%，而正常的概率应该是1/7=14.28%,也就是第一个节点被选中的概率小于正常概率，所以这个算法不准去

**Reply by y761823** [reply]
第一个节点可以被交换到别的地方，然后后面可能又会被交换回第一个位置。确实有问题，不能这么算

**Reply by y761823** [reply]
| 剩余循环次数 | 当前0留在首位，最后0在首位的概率 | 当前0不在首位，最后0在首位的概率 |
| ---- | ------ | ------ |
| 0 | 1 | 0 |
| 1 | 0.755102041 | 0.040816327 |
| 2 | 0.580174927 | 0.069970845 |
| 3 | 0.455226989 | 0.090795502 |
| 4 | 0.365978461 | 0.105670256 |
| 5 | 0.302229513 | 0.116295081 |
| 6 | 0.25669455 | 0.123884242 |
| 7 | 0.224169577 | 0.129305071 |

假设list里是 0 1 2 3 4 5 6 7
一次循环中，0在首位，0不会被换走的概率 x = (6/7)^2+(1/7)^2
一次循环中，0不在首位，0会被换回首位的概率 y = 2/(7^2)
大概这么dp一下，跟验证的结果一样，我只会dp，不知道有没有什么数学一点的办法
